File: 5 Testen und Debugging/(0) HeizungProjekt_V0/Heizung.java
/**
 * Die Klasse Heizung simuliert das Verhalten einer Heizung.
 * Autor:   Martin Lentz
 * Version: 10.12.2016
 */
public class Heizung
{
    // Eigenschaften
    private int temperatur, min, max, schrittweite;

    // Konstruktoren für Objekte der Klasse Heizung
    public Heizung()
    {
        min = 10;
        max = 30;
        temperatur = 15;
        setzeSchrittweite(5);
    }

    public Heizung(int neueTemperatur, int neuesMin, int neuesMax, int neueSchrittweite)
    {
        temperatur = neueTemperatur;
        min = neuesMin;
        max = neuesMax;
        setzeSchrittweite(neueSchrittweite);
    }

    // Liefert den aktuellen Temperaturwert
    public int gibTemperatur()
    {
        return temperatur;
    }

    // Liefert die aktuelle Schrittweite
    public int gibSchrittweite()
    {
        return schrittweite;
    }

    // Setzt die Schrittweite
    public void setzeSchrittweite(int neueSchrittweite)
    {
        if (neueSchrittweite > 0)
        {
            schrittweite = neueSchrittweite;
        }
        else if (neueSchrittweite < 0)
        {
            schrittweite = -neueSchrittweite;
        }
        else
        {
            schrittweite = 5;
        }
    }   

    // Erhöhen der Temperatur
    public void waermer()
    {
        if (temperatur+schrittweite <= max)
        {
            temperatur = temperatur+schrittweite;
        }
        else
        {
            temperatur = max;
        }
    }

    // Absenken der Temperatur
    public void kuehler()
    {
        if (temperatur-schrittweite >= min)
        {
            temperatur = temperatur-schrittweite;
        }
        else
        {
            temperatur = min;
        }
    }
}

File: Informatik/5 Testen und Debugging/HeizungProjektCSharp/HeizungProjektCSharp/Heizung.cs
namespace HeizungProjektCSharp;

public class Heizung
{
	private readonly int _min;
	private readonly int _max;
	private int _schrittweite;
	private int _temperatur;

	public int Temperatur
	{
		get => _temperatur;
		private set
		{
			if (value < _min) _temperatur = _min;
			else if (value > _max) _temperatur = _max;
			else _temperatur = value;
		}
	}

	public int Schrittweite
	{
		get => _schrittweite;
		set
		{
			if (value < 0) _schrittweite = -value;
			else if (value == 0) _schrittweite = 5;
			else _schrittweite = value;
		}
	}

	public Heizung(int temperatur = 15, int min = 10, int max = 30, int schrittweite = 5)
	{
		_min = min;
		_max = max;
		_schrittweite = schrittweite;
		Temperatur = temperatur;
	}

	public void Waermer()
	{
		Temperatur += _schrittweite;
	}
	
	public void Kuehler()
	{
		Temperatur -= _schrittweite;
	}
}
